spanner/spanner_snippets/spanner/integration_test.go

```diff
@@ -582,6 +583,53 @@ func TestCreateDatabaseWithDefaultLeaderSample(t *testing.T) {
 	assertContains(t, out, "The result of the query to get")
 }
 
+func TestUpdateDatabaseSample(t *testing.T) {
+	_ = testutil.SystemTest(t)
+	t.Parallel()
+
+	_, dbName, cleanup := initTest(t, randomID())
+	defer cleanup()
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Hour)
+	defer cancel()
+
+	var out string
+	mustRunSample(t, createDatabase, dbName, "failed to create a database")
+	out = runSampleWithContext(ctx, t, updateDatabase, dbName, "failed to update database")
+	assertContains(t, out, fmt.Sprintf("Updated database [%s]\n", dbName))
+
+	databaseAdmin, err := database.NewDatabaseAdminClient(ctx)
+	if err != nil {
+		log.Fatalf("cannot create databaseAdmin client: %v", err)
+	}
+
+	database, err := databaseAdmin.GetDatabase(ctx, &adminpb.GetDatabaseRequest{Name: dbName})
+	if err != nil {
+		log.Fatalf("error during GetDatabase for db %v: %v", dbName, err)
+	}
+	// Verify if the EnableDropProtection field got enabled
+	if !database.GetEnableDropProtection() {
+		t.Errorf("got output %t; want it to contain %t", database.GetEnableDropProtection(), true)
+	}
+
+	// Disable Drop db protection for the cleanup to delete the databases
```

Comment:
Perhaps the overall test cleanup should do this and retry in case of a transient failure?

Reply:
Thanks for the explanation... two thoughts on this:
1. Maybe the logic to set `EnableDropProtection: false` should go in the cleanup function rather than the test itself?
1. I think it'd be a good idea to ensure that the call to `databaseAdmin.UpdateDatabase` is itself retried in cases of transient errors to make sure you don't end up with stranded resources in the test project.

Both of these are up to you if you want to implement, just suggestions.